bestofjs/bestofjs
# ESLint recommended setup, cleanup of errors and warnings

## Goal

- Add TypeScript ESLint recommended setup (to catch basic problems such as unused variables)
- Fix ESLint errors and warnings
- Remove `eslint-plugin-unused-imports` plugin as the feature is already included by the TypeScript ESLint recommended set
- Remove warning when running commands from the PNPM root level

```
The field "pnpm" was found in /Users/michaelrambeau/dev/bestofjs/bestofjs-monorepo/apps/bestofjs-webui/package.json. This will not take effect. You should configure "pnpm" at the root of the workspace instead
```

This PR also adds a basic workflow on GitHub actions to ensure `test` and `lint` command pass, before a PR is merged.

## How to test

- Run `pnpm install` at the root level
- check the absence of warnings
- Run `pnpm -F bestofjs-nextjs lint` there should be no errors and no warnings

## Screenshots

Warning removed:

![image](https://github.com/bestofjs/bestofjs/assets/5546996/ba8d986f-2d58-4f8f-a9e4-73d7dbcb019a)

Lint command:

![image](https://github.com/bestofjs/bestofjs/assets/5546996/df9b88c6-a2ac-44f8-9cd4-b1282730c6eb)
